test/validator/strict_dependencies_test.dart

```diff
@@ -172,5 +199,43 @@ main() {
       ]).create();
       expectValidationWarning(strictDeps);
     });
+
+    group('declares an import with an invalid package URL: ', () {
+      integration('"package:"', () {
+        d.dir(appPath, [
+          d.dir('lib', [
+            d.file('library.dart', r'''
+            import 'package:';
+          '''),
+          ]),
+        ]).create();
+        expectValidationWarning(strictDeps);
+      });
+
+      integration('"package:silly_monkey"', () {
+        d.dir(appPath, [
+          d.libPubspec("test_pkg", "1.0.0", deps: {
+            "silly_monkey": "^1.2.3"
+          }, sdk: ">=1.8.0 <2.0.0"),
+          d.dir('lib', [
+            d.file('library.dart', r'''
+            import 'package:silly_monkey';
+          '''),
+          ]),
+        ]).create();
+        expectValidationWarning(strictDeps);
+      });
+
+      integration('"package:/"', () {
+        d.dir(appPath, [
+          d.dir('lib', [
+            d.file('library.dart', r'''
+            import 'package:/';
+          '''),
+          ]),
+        ]).create();
+        expectValidationWarning(strictDeps);
+      });
+    });
```

Comment:
We still need to test for `package:/]`. This is meaningfully different because it's invalid URL syntax, whereas the rest of these are invalid only as `package:` URLs in particular.